terraform-kinesis-poison-pill/README.md

```diff
@@ -0,0 +1,149 @@
+# Kinesis Poison Pill Pattern
+
+This pattern demonstrates how to handle a Lambda consumer failure when reading from a Kinesis Data Stream with Terraform
+
+Without proper handling of failure when working with Kinesis Data Streams, an iterator will get stuck and the only way for the data to clear the stream is for it to **Age Out** beyond the trim horizon.
+
+This will ultimately create wasteful invocations of a Lambda, wasted CPU cycles in a container and worst of all the downstream consumers will not get the data they need.
+
+Learn more about this pattern at Serverless Land Patterns: << Add the live URL here >>
+
+**Important**: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+-   [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+-   [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+-   [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+-   [Node and NPM](https://nodejs.org/en/download/) installed
+-   [The Go Programming Langage](https://go.dev/doc/install) must be installed in order to build the Lambda
+-   [Terraform](https://www.terraform.io) must be installed in order to provision infrastructure
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ```
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the lambda module directory:
+    ```
+    cd terraform-kinesis-poison-pill/modules/lambda
+    ```
+3. Install the project dependencies 
+    ```
+     npm install
+    ```
+4. Build lambda using make command
+    ```
+     make
+    ```
+5. Change directory to the pattern directory:
+    ```
+    cd ../..
+
+6. Deploy the stack to your default AWS account and region using terraform
+    ```
+    terraform init
+    terraform plan
+    terraform apply
+    ```
+
+## How it works
+
+This pattern is designed to highlight how to handle failure when processing records in a Lambda. The `Kinesis Event Source` turns a Lambda into a Data Consumer which will process a JSON event that looks like this in Go
+
+```golang
```

Comment:
when this part is visualized all until the end is switching in weird places from code to no code. please fix

Reply:
I am able to preview it correct, can you elaborate or share screen shot please.

<img width="1057" alt="image" src="https://github.com/aws-samples/serverless-patterns/assets/57580340/f8bb00ab-fb9a-44b4-9c5f-441d8c3a6450">